src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseStatistics.hpp

```diff
@@ -61,67 +61,188 @@ class CompositionalMultiphaseStatistics : public FieldStatisticsBase< Compositio
                         real64 const eventProgress,
                         DomainPartition & domain ) override;
 
-  /**@}*/
-
-private:
-
-  using Base = FieldStatisticsBase< CompositionalMultiphaseBase >;
-
-  /**
-   * @struct viewKeyStruct holds char strings and viewKeys for fast lookup
-   */
-  struct viewKeyStruct
+  class RegionStatistics : public dataRepository::Group
   {
-    /// String for the flag deciding the computation of the CFL numbers
-    constexpr static char const * computeCFLNumbersString() { return "computeCFLNumbers"; }
-    /// String for the flag deciding the computation of the region statistics
-    constexpr static char const * computeRegionStatisticsString() { return "computeRegionStatistics"; }
-    /// String for the region statistics
-    constexpr static char const * regionStatisticsString() { return "regionStatistics"; }
-    /// String for the relperm threshold
-    constexpr static char const * relpermThresholdString() { return "relpermThreshold"; }
-  };
+public:
+    RegionStatistics( string const & name,
+                      Group * const parent )
+      : Group( name, parent )
+    {
+
+      registerWrapper( viewKeyStruct::averagePressureString(), &m_averagePressure ).
+        setApplyDefaultValue( 0 ).
+        //setInputFlag( dataRepository::InputFlags::OPTIONAL ).
+        setDescription( "average region pressure" );
```

Comment:
That is way too long to go in the declaration. And as it is not templated, it goes in the `.cpp`.